### ISSUE TITLE
[State Magement] Include setState + new context API

### ISSUE THREAD

**Issue by aoc** [adam-golab/react-developer-roadmap#4]
I think it's important to fully understand setState and new context API before learning a more complex solution like redux or mobX.
It usually covers most of the state management requirements and let's you understand what problems those tool help you with.

**Comment by aoc** [comment]
great work btw! :) I can create a PR if you want

**Comment by adam-golab** [comment]
Thanks for adding this issue with your thoughts :)
Regarding the setState and state management built into React, I think that it's included in the "Learn React" step. It may be a good idea to explicitly emphasize that learning React contains following key features: 
- class Component, stateless, PureComponent
- state management (setState)
- lifecycle methods
- JSX
- context API
- and so on (if you have any additional key features, feel free to suggest)
I think the best place for such information is not on the graphic but in the readme. 

Regarding the context API, to be honest I don't agree that is necessary to understand it before learning  Redux or Mobx. It's very helpful to know this mechanism, it explains a lot of things that happen "under the hood". But the knowledge that redux's `Provider` component and `connect` HOC can pass state through every level of the tree is sufficient at the beginnings.

**Comment by aoc** [comment]
I didn't mean that the context API is required to understand redux / mobx, and I agree with you it's not necessary.

Looking at the graphic someone may incorrectly think you require a library to handle state management, specially if you're talking about learning material.

In my opinion you can introduce the concept of state management (local state vs global state, lifting state up, etc) only using React's API. After that you can explain which problems redux and mobx try to solve.

Passing state through different levels of the tree can be explained using the new context API.

**Comment by aoc** [comment]
Maybe it's a matter of taste, but I try to avoid introducing libraries, which include more complexity and new problems. Of course there is always a tradeoff, but you need to fully understand the underlying problem (or limitation) the library is trying to solve to decide to decide if it's worth it.

**Comment by adam-golab** [comment]
Yeah, I agree with you. There is no need to learn any additional libraries to manage React's state. 

So what's your suggestion? Should I add an additional block next to `Redux` and `MobX`?

**Comment by aoc** [comment]
Yes, i think it's only a clarification
